scripts/pi-hole/php/teleporter.php

```diff
@@ -118,40 +106,43 @@ function getWildcardListContent() {
 			}
 		}
 
-		$continue = strtolower($name[1]) == 'zip' ? true : false;
+		$continue = strtolower($name[1]) == 'tar' && strtolower($name[2]) == 'gz' ? true : false;
 		if(!$continue || !$okay) {
-			die("The file you are trying to upload is not a .zip file (filename: ".$filename.", type: ".$type."). Please try again.");
+			die("The file you are trying to upload is not a .tar.gz file (filename: ".$filename.", type: ".$type."). Please try again.");
```

Comment:
Possible XSS attack vector (`$filename` and `$type` being injected into the page)